scripts/export_database.pl

```diff
@@ -298,6 +303,12 @@ sub sanitize_field_content {
 			my $url = "http://world-$lc.$server_domain" . product_url($product_ref);
 			my $code = ($product_ref->{code} // '');
 
+			if ($product_ref->{obsolete}) {
+				if ($exported_products{$code}) {
+					next;
+				}
```

Comment:
The idea is correct but you must define $exported_products variable (`my %exported_products = ()`)